src/chainparams.cpp

```diff
@@ -990,7 +990,293 @@ class CLiquidV1Params : public CChainParams {
+
+        if (args.IsArgSet("-parentgenesisblockhash")) {
+            parentGenesisBlockHash = uint256S(args.GetArg("-parentgenesisblockhash", ""));
+        }
+        // Either it has a parent chain or not
+        const bool parent_genesis_is_null = parentGenesisBlockHash == uint256();
+        assert(consensus.has_parent_chain != parent_genesis_is_null);
+        if (args.IsArgSet("-con_parentpowlimit")) {
+            consensus.parentChainPowLimit = uint256S(args.GetArg("-con_parentpowlimit", ""));
+        }
+
+        if (args.IsArgSet("-con_parent_chain_signblockscript")) {
+            consensus.parent_chain_signblockscript = StrHexToScriptWithDefault(args.GetArg("-con_parent_chain_signblockscript", ""), CScript());
+        }
+        consensus.pegin_min_depth = args.GetArg("-peginconfirmationdepth", consensus.pegin_min_depth);
+
+        if (args.IsArgSet("-fedpegscript")) {
+            consensus.fedpegScript = StrHexToScriptWithDefault(args.GetArg("-fedpegscript", ""), CScript());
+        }
+
+        consensus.total_valid_epochs = args.GetArg("-total_valid_epochs", consensus.total_valid_epochs);
+
+        // Calculate pegged Bitcoin asset
+        std::vector<unsigned char> commit = CommitToArguments(consensus, strNetworkID);
+        uint256 entropy;
+        GenerateAssetEntropy(entropy,  COutPoint(uint256(commit), 0), parentGenesisBlockHash);
+        CalculateAsset(consensus.pegged_asset, entropy);
+
+        if (args.IsArgSet("-con_parent_pegged_asset")) {
+            consensus.parent_pegged_asset.SetHex(args.GetArg("-con_parent_pegged_asset", ""));
+        }
+        initial_reissuance_tokens = args.GetArg("-initialreissuancetokens", initial_reissuance_tokens);
+
+        if (args.IsArgSet("-subsidyasset")) {
+            consensus.subsidy_asset = CAsset(uint256S(args.GetArg("-subsidyasset", "")));
+        }
+
+        if (args.IsArgSet("-con_dyna_deploy_start")) {
+            consensus.vDeployments[Consensus::DEPLOYMENT_DYNA_FED].bit = 25;
+            consensus.vDeployments[Consensus::DEPLOYMENT_DYNA_FED].nStartTime = args.GetArg("-con_dyna_deploy_start", Consensus::BIP9Deployment::ALWAYS_ACTIVE);
+            consensus.vDeployments[Consensus::DEPLOYMENT_DYNA_FED].nTimeout = Consensus::BIP9Deployment::NO_TIMEOUT;
+        }
+
+        // END ELEMENTS fields
+
+        // CSV always active by default, unlike regtest
+        if (args.IsArgSet("-con_csv_deploy_start")) {
+            consensus.vDeployments[Consensus::DEPLOYMENT_CSV].bit = 0;
+            consensus.vDeployments[Consensus::DEPLOYMENT_CSV].nStartTime = args.GetArg("-con_csv_deploy_start", Consensus::BIP9Deployment::ALWAYS_ACTIVE);
+            consensus.vDeployments[Consensus::DEPLOYMENT_CSV].nTimeout = Consensus::BIP9Deployment::NO_TIMEOUT;
+        }
+    }
+
+    // XXX: This is copy-and-pasted from CCustomParams; sharing it would be better, but is annoying.
+    void SetGenesisBlock() {
```

Comment:
Since this is a clone of liquidv1, we can instead assert that `genesis_style` is `"elements"` and only have that part be in this method.